### ISSUE TITLE
Amazon Appstore support

### ISSUE THREAD

**Issue by jakubvimn** [Triple-T/gradle-play-publisher#583]
### Problem description

The plugin can be used only for Google Play, but it doesn't support Amazon Appstore

### Potential solutions/workarounds

Add the support to it by wrapping the Amazon App Submission API

### Additional context

https://developer.amazon.com/docs/app-submission-api/overview.html

**Comment by SUPERCILEX** [comment]
I have some long-term plans for this, so thanks for opening the issue. 👍

**Comment by TakuSemba** [comment]
I want this too.

**Comment by AndreSand** [comment]
I will use the feature to publish to Amazon store :)

**Comment by BrantApps** [comment]
Just so you know I built a plugin inspired by this one to automate my publishing to the Amazon App Store. @SUPERCILEX I'm not sure how much is PR'able to be able to merge into this product's functionality but do take a look if you'd like to cargo anything.

**Comment by ScottPierce** [comment]
Would be really nice to have this.

**Comment by timbyu** [comment]
I am so anxious for this I am willing to work on the ticket.  The documentation provided is great for publishing.  Is there any documentation for extending this tool?  Or suggestions for what files I should modify to add this functionality?

Tim

**Comment by ScottPierce** [comment]
I feel that Amazon's documentation around this API is very poor (as most of their developer documentation is). I would have happily done this work a few months ago, but I couldn't even find the proper way.

**Comment by SUPERCILEX** [comment]
Feel free to give it a go. The first step would be to create a directory called `amazon/plugin` and stand up a new plugin in there. Then you'd want to add a proof of concept APK publishing task. After that pull out the API layer into `amazon/amazon-publisher` and finally we'd want to figure out how to make the plugin production ready with code reuse and tests. Note that I can't find an SDK for the amazon API which would be a huge bummer of it doesn't exist.

**Comment by SUPERCILEX** [comment]
> I couldn't even find the proper way

Yeah, I think you'd have to use retrofit or something and manually connect the dots.

**Comment by BrantApps** [comment]
I have a working gradle plugin that uploads binaries to the Amazon Developer Console to help you get a feel of how to connect the dots @ScottPierce & @timbyu!

[Here](https://github.com/BrantApps/gradle-amazon-app-store-publisher) it is.

The general gist of it can be gleaned from the console output the plugin generates...

![20210624_213054.jpg](https://user-images.githubusercontent.com/1109992/123328645-be702100-d533-11eb-89c2-a3e1f8402adb.jpg)

The code is quite boring. I hadn't written much Kotlin at the time either so probably looks like very Java-ry Kotlin but easy to evolve if you want to bootstrap.

**Comment by timbyu** [comment]
The Amazon documentation can be found here [https://developer.amazon.com/docs/app-submission-api/overview.html](https://developer.amazon.com/docs/app-submission-api/overview.html).  Using Brant's solution (Thanks Brant) I was successfully able to upload the APK to Amazon but I would like to use this tool as a single solution for both stores.  I am not experienced with Kotlin so it will take me a while to get it up and running.  If any of you with experience are willing to look at Brant's solution as an example I will hold off.  If not, I will give it a go.  Thanks for the quick responses!

**Comment by ursusursus** [comment]
People will want this since Windows 11 will include the amazon app store

**Comment by kenyee** [comment]
Might as well ask for the Samsung App Store too if the upload API can be generalized :-)

**Comment by joshuamking** [comment]
The Samsung App Store would seriously be amazing..

**Comment by dg76** [comment]
No, the Amazon App Store is not closed. As you can read on the website that you referenced:

> [What happens to my apps after the discontinuation of the Amazon Appstore on Android?](javascript:void(0))
> Starting August 20, 2025, any apps downloaded from the Amazon Appstore will not be guaranteed to operate on Android devices.
> **Amazon Appstore will continue to be available elsewhere, including on Fire TV and Fire Tablet devices.**
